### PR TITLE
Update to latest async lightbulb api

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/Suggestions/AsyncSuggestedActionsSource.cs [hunk 1]
```diff
@@ -36,20 +38,40 @@ private partial class AsyncSuggestedActionsSource : SuggestedActionsSource, ISug
...
+                try
+                {
+                    await GetSuggestedActionsWorkerAsync(
+                        requestedActionCategories, range, collectors, cancellationToken).ConfigureAwait(false);
+                }
+                finally
+                {
+                    // Always ensure that all the collectors are marked as complete so we don't hang the UI.
+                    foreach (var collector in collectors)
+                        collector.Complete();
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Suggestions/AsyncSuggestedActionsSource.cs:57]: @AmadeusW i hope this is ok.  First, i really don't want to have to keep track of what collectors i already called Complete on.  Second, it feels weird that the callee needs to do this.  the caller can just move all collectors to the completed state once the async call has completed.

File: src/Workspaces/Core/Portable/Experiments/IExperimentationService.cs [hunk 2]
```diff
@@ -39,7 +39,6 @@ internal static class WellKnownExperimentNames
         public const string OOPServerGC = "Roslyn.OOPServerGC";
         public const string LspTextSyncEnabled = "Roslyn.LspTextSyncEnabled";
         public const string ImportsOnPasteDefaultEnabled = "Roslyn.ImportsOnPasteDefaultEnabled";
-        public const string LspTextSyncEnabled = "Roslyn.LspTextSyncEnabled";
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Experiments/IExperimentationService.cs:42]: some bad earlier merge caused this.  it's already defined two lines above, so this was causing a build break

File: src/EditorFeatures/Core.Wpf/Suggestions/AsyncSuggestedActionsSource.cs (hunk 1 above)
Comment (AmadeusW) [reply src/EditorFeatures/Core.Wpf/Suggestions/AsyncSuggestedActionsSource.cs:57]: That's not ok, calling Complete causes collector to send its SuggestedActionSets to be displayed. Note that when actions are displayed, they can't be reordered. The correct approach is to call .Add to add SuggestedActionSets for a given priority, and call .Complete when all actions for this priority are sent. This signals Editor to display these actions in the UI.

File: src/EditorFeatures/Core.Wpf/Suggestions/AsyncSuggestedActionsSource.cs [hunk 3]
```diff
@@ -19,12 +20,14 @@
 using Microsoft.VisualStudio.Text;
 using Microsoft.VisualStudio.Text.Editor;
 using Roslyn.Utilities;
+using Microsoft.CodeAnalysis.Shared.Collections;
+using Microsoft.CodeAnalysis.PooledObjects;
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.Suggestions
 {
     internal partial class SuggestedActionsSourceProvider
```
Comment (AmadeusW) [inline src/EditorFeatures/Core.Wpf/Suggestions/AsyncSuggestedActionsSource.cs:28]: `SuggestedActionsSourceProvider` needs to be decorated with `SuggestedActionPriorityAttribute` which determines the priorities you support (e.g. `DefaultOrderings.High` and `DefaultOrderings.Default`)

File: src/EditorFeatures/Core.Wpf/Suggestions/AsyncSuggestedActionsSource.cs [hunk 4]
```diff
@@ -36,20 +39,45 @@ private partial class AsyncSuggestedActionsSource : SuggestedActionsSource, ISug
...
+                using var _ = ArrayBuilder<ISuggestedActionSetCollector>.GetInstance(out var completedCollectors);
+                try
+                {
+                    await GetSuggestedActionsWorkerAsync(
+                        requestedActionCategories, range, collectors, completedCollectors, cancellationToken).ConfigureAwait(false);
+                }
+                finally
+                {
+                    // Always ensure that all the collectors are marked as complete so we don't hang the UI.
+                    foreach (var collector in collectors)
```
Comment (AmadeusW) [inline src/EditorFeatures/Core.Wpf/Suggestions/AsyncSuggestedActionsSource.cs:58]: the idea is that you complete collector as soon as you're done giving it items.
completing all collectors at once does not give users the benefit of progressively streaming in items, and makes them high priority actions wait for the slow low priority actions to compute

Comment (AmadeusW) [reply src/EditorFeatures/Core.Wpf/Suggestions/AsyncSuggestedActionsSource.cs:58]: See [this sample](https://gist.github.com/AmadeusW/6ce38855a8fb14d01af228de57ecdfb8) for inspiration

**Review by AmadeusW — CHANGES_REQUESTED** [review]

**Comment by CyrusNajmabadi** [comment]
@AmadeusW can you ptal?

**Review by AmadeusW — APPROVED** [review]
🚢💡

**Comment by CyrusNajmabadi** [comment]
@AmadeusW did that new package change other things?  It seems to have broken a bunch of tests and caused a lot of integration test timeouts :(

**Comment by AmadeusW** [comment]
You've updated quite a few versions, so you'd get other changes made by the Editor team. What exactly regressed?

**Comment by CyrusNajmabadi** [comment]
Discussed offline.  Editor has broken enum-preselection.  Tracking issue is here: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1363210/

**Review by davidwengier — APPROVED** [review]

**Pull request merged**